### ISSUE TITLE
Incorrect warnings in class initialization scope with enabled type check

### ISSUE THREAD

**Issue by m4tiz5zktonj** [markw65/monkeyc-optimizer#35]
```monkey-c
import Toybox.Lang;

class Test {
    static const CONST_1 = (1).toNumber();
    // Unexpected types for operator '+': [Number vs Null]
    static const CONST_2 = 1 + (0.5 * 2).toNumber();

    var a as Number;
    var b as Number;
    var c as Number;

    function initialize(x as Number) {
        a = x;
        b = getValue(a);
        // Argument 1 to $.Test.getValue expected to be Number but got Null or Number
        c = getValue(a);
    }

    function getValue(x as Number) as Number {
        // Unexpected types for operator '*': [Number vs Null]
        return x * CONST_1;
    }
}
```

**Comment by markw65** [comment]
> One more in 2.0.97, sorry

I guess I should just fix this properly, or you'll just keep extending the test case :-)

The issue is that I designed type propagation to work function-at-a-time, and I don't have a function in these cases, but I figured I could get away with ad hoc analysis because the expressions would be simple.

I should just synthesize a method for all the const/var initializers in any given scope, and then run the normal type propagation on that. Which is pretty much what Garmin does in the runtime anyway - it puts all those initializers into the `<init>` method.
